scripts/benchmarks_monthly.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+
+'''
+This script provides simple execution of the .NET micro benchmarks
+for a .NET preview release, making it easier to contribute to our
+monthly manual performance runs.
+'''
+
+from performance.logger import setup_loggers
+from argparse import ArgumentParser, ArgumentTypeError
+from datetime import datetime
+from logging import getLogger
+
+import benchmarks_ci
+import tarfile
+import shutil
+import sys
+import os
+
+VERSIONS = {
+    'net7.0-preview2': { 'tfm': 'net7.0', 'build': '7.0.100-preview.2.22124.4' },
+    'net7.0-preview1': { 'tfm': 'net7.0', 'build': '7.0.100-preview.1.22077.12' },
+    'net6.0': { 'tfm': 'net6.0' }
+}
+
+def get_version_from_name(name: str) -> str:
+    for version in VERSIONS:
+        if version == name:
+            return VERSIONS[version]
+
+    raise Exception('The version specified is not supported', name)
+
+def add_arguments(parser: ArgumentParser) -> ArgumentParser:
+    # Adds new arguments to the specified ArgumentParser object.
+
+    if not isinstance(parser, ArgumentParser):
+        raise TypeError('Invalid parser.')
+
+    parser.add_argument(
+        'versions',
+        nargs='+',
+        choices=VERSIONS,
+        help='Specifies the .NET version(s) for the benchmarks run')
+
+    parser.add_argument(
+        '--device-name',
+        dest='device_name',
+        help='The name of this device, to be included in the .tar.gz file name')
+
+    parser.add_argument(
+        '--filter',
+        dest='filter',
+        default='*',
+        help='Specifies the benchmark filter to pass to BenchmarkDotNet')
+
+    parser.add_argument(
+        '--architecture',
+        dest='architecture',
+        choices=['x64', 'x86', 'arm64', 'arm'],
+        default='x64',
+        help='Specifies the SDK processor architecture')
+
+    parser.add_argument(
+        '--bdn-arguments',
+        dest='bdn_arguments',
+        help='Command line arguments to be passed to BenchmarkDotNet, wrapped in quotes',
+    )
+
+    parser.add_argument(
+        '--no-clean',
+        dest='no_clean',
+        action='store_true',
+        help='Do not clean the SDK and results directories before execution')
+
+    parser.add_argument(
+        '--dry-run',
+        dest='dry_run',
+        action='store_true',
+        help='Perform a dry run, showing what would be executed')
+
+    return parser
+
+def __process_arguments(args: list):
+    parser = ArgumentParser(
+        description='Tool to execute the monthly manual micro benchmark performance runs',
+        allow_abbrev=False
+    )
+
+    add_arguments(parser)
+    return parser.parse_args(args)
+
+def __main(args: list) -> int:
+    setup_loggers(verbose=True)
+
+    args = __process_arguments(args)
+    rootPath = os.path.normpath(os.path.join(os.path.dirname(__file__), '..'))
+    sdkPath = os.path.join(rootPath, 'tools', 'dotnet', args.architecture, 'sdk')
```

Comment:
I think it would be better to delete everything for given architecture, not just the sdk subfolder. Otherwise, the `dotnet --info` run by python script is still reporting the older versions that were previously downloaded:

```suggestion
    sdkPath = os.path.join(rootPath, 'tools', 'dotnet', args.architecture)
```

Before:

<details>

```log
PS D:\projects\performance> py .\scripts\benchmarks_monthly.py net7.0-preview1 net7.0-preview2 --filter *Burgers* --architecture x86
[2022/03/01 09:51:09][INFO] ----------------------------------------------
[2022/03/01 09:51:09][INFO] Initializing logger 2022-03-01 09:51:09.834451
[2022/03/01 09:51:09][INFO] ----------------------------------------------
[2022/03/01 09:51:09][INFO] rmdir -r .\tools\dotnet\x86\sdk
[2022/03/01 09:51:10][INFO] rmdir -r .\artifacts\bin\MicroBenchmarks\Release\net7.0\BenchmarkDotNet.Artifacts\results
[2022/03/01 09:51:10][INFO] Executing: benchmarks_ci.py --skip-logger-setup --filter *Burgers* --architecture x86 -f net7.0 --dotnet-versions 7.0.100-preview.1.22077.12
[2022/03/01 09:51:10][INFO] Installing tools.
[2022/03/01 09:51:10][INFO] ----------------------
[2022/03/01 09:51:10][INFO] Downloading DotNet Cli
[2022/03/01 09:51:10][INFO] ----------------------
[2022/03/01 09:51:10][INFO] DotNet Install Path: 'D:\projects\performance\tools\dotnet\x86'
[2022/03/01 09:51:10][INFO] Downloading https://dot.net/v1/dotnet-install.ps1
[2022/03/01 09:51:11][INFO] $ pushd "D:\projects\performance"
[2022/03/01 09:51:11][INFO] $ powershell.exe -NoProfile -ExecutionPolicy Bypass D:\projects\performance\tools\dotnet\x86\dotnet-install.ps1 -InstallDir D:\projects\performance\tools\dotnet\x86 -Architecture x86 -Version 7.0.100-preview.1.22077.12
[2022/03/01 09:51:12][INFO] dotnet-install: Note that the intended use of this script is for Continuous Integration (CI) scenarios, where:
[2022/03/01 09:51:12][INFO] dotnet-install: - The SDK needs to be installed without user interaction and without admin rights.
[2022/03/01 09:51:12][INFO] dotnet-install: - The SDK installation doesn't need to persist across multiple CI runs.
[2022/03/01 09:51:12][INFO] dotnet-install: To set up a development environment or to run apps, use installers rather than this script. Visit https://dotnet.microsoft.com/download to get the installer.
[2022/03/01 09:51:12][INFO]
[2022/03/01 09:52:04][INFO] dotnet-install: Extracting the archive.
[2022/03/01 09:52:18][INFO] dotnet-install: Adding to current process PATH: "D:\projects\performance\tools\dotnet\x86\". Note: This change will not be visible if PowerShell was run as a child process.
[2022/03/01 09:52:18][INFO] dotnet-install: Note that the script does not resolve dependencies during installation.
[2022/03/01 09:52:18][INFO] dotnet-install: To check the list of dependencies, go to https://docs.microsoft.com/dotnet/core/install/windows#dependencies
[2022/03/01 09:52:18][INFO] dotnet-install: Installation finished
[2022/03/01 09:52:18][INFO] $ popd
[2022/03/01 09:52:18][INFO] $ dotnet --info
[2022/03/01 09:52:18][INFO] .NET SDK (reflecting any global.json):
[2022/03/01 09:52:18][INFO]  Version:   7.0.100-preview.1.22077.12
[2022/03/01 09:52:18][INFO]  Commit:    34e0fba2d3
[2022/03/01 09:52:18][INFO]
[2022/03/01 09:52:18][INFO] Runtime Environment:
[2022/03/01 09:52:18][INFO]  OS Name:     Windows
[2022/03/01 09:52:18][INFO]  OS Version:  10.0.22000
[2022/03/01 09:52:18][INFO]  OS Platform: Windows
[2022/03/01 09:52:18][INFO]  RID:         win10-x86
[2022/03/01 09:52:18][INFO]  Base Path:   D:\projects\performance\tools\dotnet\x86\sdk\7.0.100-preview.1.22077.12\
[2022/03/01 09:52:18][INFO]
[2022/03/01 09:52:18][INFO] Host (useful for support):
[2022/03/01 09:52:18][INFO]   Version: 7.0.0-preview.2.22122.4
[2022/03/01 09:52:18][INFO]   Commit:  1615de6ccc
[2022/03/01 09:52:18][INFO]
[2022/03/01 09:52:18][INFO] .NET SDKs installed:
[2022/03/01 09:52:18][INFO]   7.0.100-preview.1.22077.12 [D:\projects\performance\tools\dotnet\x86\sdk]
[2022/03/01 09:52:18][INFO]
[2022/03/01 09:52:18][INFO] .NET runtimes installed:
[2022/03/01 09:52:18][INFO]   Microsoft.AspNetCore.App 6.0.2 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.AspNetCore.App]
[2022/03/01 09:52:18][INFO]   Microsoft.AspNetCore.App 7.0.0-preview.1.22077.2 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.AspNetCore.App]
[2022/03/01 09:52:18][INFO]   Microsoft.AspNetCore.App 7.0.0-preview.2.22107.8 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.AspNetCore.App]
[2022/03/01 09:52:18][INFO]   Microsoft.AspNetCore.App 7.0.0-preview.3.22117.6 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.AspNetCore.App]
[2022/03/01 09:52:18][INFO]   Microsoft.NETCore.App 6.0.2 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.NETCore.App]
[2022/03/01 09:52:18][INFO]   Microsoft.NETCore.App 7.0.0-preview.1.22076.8 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.NETCore.App]
[2022/03/01 09:52:18][INFO]   Microsoft.NETCore.App 7.0.0-preview.2.22103.2 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.NETCore.App]
[2022/03/01 09:52:18][INFO]   Microsoft.NETCore.App 7.0.0-preview.2.22122.4 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.NETCore.App]
[2022/03/01 09:52:18][INFO]   Microsoft.WindowsDesktop.App 6.0.2 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.WindowsDesktop.App]
[2022/03/01 09:52:18][INFO]   Microsoft.WindowsDesktop.App 7.0.0-preview.1.22076.3 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.WindowsDesktop.App]
[2022/03/01 09:52:18][INFO]   Microsoft.WindowsDesktop.App 7.0.0-preview.2.22107.1 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.WindowsDesktop.App]
[2022/03/01 09:52:18][INFO]   Microsoft.WindowsDesktop.App 7.0.0-preview.2.22123.5 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.WindowsDesktop.App]
```

</details>

After:

<details>

```log
PS D:\projects\performance> py .\scripts\benchmarks_monthly.py net7.0-preview1 net7.0-preview2 --filter *Burgers* --architecture x86
[2022/03/01 13:37:30][INFO] ----------------------------------------------
[2022/03/01 13:37:30][INFO] Initializing logger 2022-03-01 13:37:30.601280
[2022/03/01 13:37:30][INFO] ----------------------------------------------
[2022/03/01 13:37:30][INFO] rmdir -r .\tools\dotnet\x86
[2022/03/01 13:37:31][INFO] rmdir -r .\artifacts\bin\MicroBenchmarks\Release\net7.0\BenchmarkDotNet.Artifacts\results
[2022/03/01 13:37:31][INFO] Executing: benchmarks_ci.py --skip-logger-setup --filter *Burgers* --architecture x86 -f net7.0 --dotnet-versions 7.0.100-preview.1.22077.12
[2022/03/01 13:37:31][INFO] Installing tools.
[2022/03/01 13:37:31][INFO] ----------------------
[2022/03/01 13:37:31][INFO] Downloading DotNet Cli
[2022/03/01 13:37:31][INFO] ----------------------
[2022/03/01 13:37:31][INFO] DotNet Install Path: 'D:\projects\performance\tools\dotnet\x86'
[2022/03/01 13:37:31][INFO] Downloading https://dot.net/v1/dotnet-install.ps1
[2022/03/01 13:37:32][INFO] $ pushd "D:\projects\performance"
[2022/03/01 13:37:32][INFO] $ powershell.exe -NoProfile -ExecutionPolicy Bypass D:\projects\performance\tools\dotnet\x86\dotnet-install.ps1 -InstallDir D:\projects\performance\tools\dotnet\x86 -Architecture x86 -Version 7.0.100-preview.1.22077.12
[2022/03/01 13:37:32][INFO] dotnet-install: Note that the intended use of this script is for Continuous Integration (CI) scenarios, where:
[2022/03/01 13:37:32][INFO] dotnet-install: - The SDK needs to be installed without user interaction and without admin rights.
[2022/03/01 13:37:32][INFO] dotnet-install: - The SDK installation doesn't need to persist across multiple CI runs.
[2022/03/01 13:37:32][INFO] dotnet-install: To set up a development environment or to run apps, use installers rather than this script. Visit https://dotnet.microsoft.com/download to get the installer.
[2022/03/01 13:37:32][INFO]
[2022/03/01 13:38:22][INFO] dotnet-install: Extracting the archive.
[2022/03/01 13:38:36][INFO] dotnet-install: Adding to current process PATH: "D:\projects\performance\tools\dotnet\x86\". Note: This change will not be visible if PowerShell was run as a child process.
[2022/03/01 13:38:36][INFO] dotnet-install: Note that the script does not resolve dependencies during installation.
[2022/03/01 13:38:36][INFO] dotnet-install: To check the list of dependencies, go to https://docs.microsoft.com/dotnet/core/install/windows#dependencies
[2022/03/01 13:38:36][INFO] dotnet-install: Installation finished
[2022/03/01 13:38:36][INFO] $ popd
[2022/03/01 13:38:36][INFO] $ dotnet --info
[2022/03/01 13:38:36][INFO] .NET SDK (reflecting any global.json):
[2022/03/01 13:38:36][INFO]  Version:   7.0.100-preview.1.22077.12
[2022/03/01 13:38:36][INFO]  Commit:    34e0fba2d3
[2022/03/01 13:38:36][INFO]
[2022/03/01 13:38:36][INFO] Runtime Environment:
[2022/03/01 13:38:36][INFO]  OS Name:     Windows
[2022/03/01 13:38:36][INFO]  OS Version:  10.0.22000
[2022/03/01 13:38:36][INFO]  OS Platform: Windows
[2022/03/01 13:38:36][INFO]  RID:         win10-x86
[2022/03/01 13:38:36][INFO]  Base Path:   D:\projects\performance\tools\dotnet\x86\sdk\7.0.100-preview.1.22077.12\
[2022/03/01 13:38:36][INFO]
[2022/03/01 13:38:36][INFO] Host (useful for support):
[2022/03/01 13:38:36][INFO]   Version: 7.0.0-preview.1.22076.8
[2022/03/01 13:38:36][INFO]   Commit:  405337939c
[2022/03/01 13:38:36][INFO]
[2022/03/01 13:38:36][INFO] .NET SDKs installed:
[2022/03/01 13:38:36][INFO]   7.0.100-preview.1.22077.12 [D:\projects\performance\tools\dotnet\x86\sdk]
[2022/03/01 13:38:36][INFO]
[2022/03/01 13:38:36][INFO] .NET runtimes installed:
[2022/03/01 13:38:36][INFO]   Microsoft.AspNetCore.App 7.0.0-preview.1.22077.2 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.AspNetCore.App]
[2022/03/01 13:38:36][INFO]   Microsoft.NETCore.App 7.0.0-preview.1.22076.8 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.NETCore.App]
[2022/03/01 13:38:36][INFO]   Microsoft.WindowsDesktop.App 7.0.0-preview.1.22076.3 [D:\projects\performance\tools\dotnet\x86\shared\Microsoft.WindowsDesktop.App]
[2022/03/01 13:38:36][INFO]
```

</details>